Mus 110 mark notes

fwinkelbauer opened the pull request.

Commit:
MUS-110 Touch Detector

Added a new class: given a list of RectFs and a touch (x, y) it can tell
you if the touch landed on a RectF object

Commit:
MUS-110 TouchDetector

Improved detector tests

Commit:
MUS-110 Marking Symbols

Changed Symbol from an interface to an abstract class. Added the marked
boolean flag.

Commit:
MUS-110 Extending draw logic

Changed the way symbols are drawn so that a marked symbol has a
different color than a normal one. NOTE: this does not yet apply to
bitmaps

Commit:
MUS-110 Renamed TrackDrawer

Renamed TrackDrawer into SymbolsDrawer (and changed parameters)

Commit:
MUS-110 Renamed TrackDrawer

Commit:
Revert "MUS-110 Renamed TrackDrawer"

This reverts commit 149352e7c1771c80947b8dbf62d0a1d3d569bee6.

Commit:
MUS-110 Renamed TrackDrawer

Renamed TrackDrawer into SymbolsDrawer (and changed parameters)

Commit:
MUS-110 Refactored NotePositionInformation

We are going to use this object for every symbol.
- Renamed class into SymbolCoordinates
- Refactored tests

Commit:
MUS-110 Refactored NotePositionInformation

Commit:
MUS-110 Refactored SymbolCoordinates

Changed the constructor

Commit:
MUS-110 Refactored Draw

NoteSheetDrawer can now return a list of coordinates of all drawn
symbols (breaks and notes). This can be used by the NoteSheetView for
marking certain notes on touch.

Commit:
MUS-110 Fixed Marking Bug

missed a negation...

Commit:
MUS-110 Redraw

- Added redraw behaviour
- Restored iml file after changing it in a previous commit

Commit:
MUS-110 Refactoring + Tests

- Added some tests
- Renamed SymbolCoordinates into SymbolPosition

Commit:
MUS-110 Refactoring + Tests

Commit:
MUS-110 Tests

added NoteSheetView tests for the touch event

Commit:
MUS-110 Refactoring

Small refactoring before the pull request

bteufl commented on musicdroidSourceTest/musicdroidSourceTest.iml:
```diff
@@ -1,32 +1,892 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
.iml file on github?! Should be removed by .gitignore as far as I know?

Commit:
MUS-110 reverting iml file

this is the current iml file at the develop branch. I am including it
because of some mistakes in prior commits.

bteufl commented on musicdroid/src/main/java/org/catrobat/musicdroid/pocketmusic/instrument/noteSheet/NoteSheetView.java:
```diff
@@ -26,29 +26,43 @@
+    protected TrackToSymbolsConverter trackConverter;
+    protected List<Symbol> symbols;
+    protected MusicalKey key;
+    protected List<SymbolPosition> symbolPositions;
```
The two lists "symbols" and "symbolPositions" have - as far as I understand - always to be synchronous. I think we should create a new data-structure for that so that this is ensured.

bteufl replied:
Gets refactored in another issue
